core/modules/node/node.pages.inc

```diff
@@ -40,8 +40,14 @@ function node_add_page() {
       );
     }
   }
-  // Bypass the node/add listing if only one content type is available.
-  if (count($content) == 1) {
+  // If there is more than one content type, sort them by title instead of
+  // machine name before displaying the list of links...
+  if (count($content) !== 1) {
+    backdrop_sort($content, array('title'));
+  }
+  // ...otherwise bypass the node/add listing and go directly to the
+  // node/add/[type] page of the single content type.
+  else {
```

Comment:
I think that this should be changed to `elseif (count($content) == 1)` as per @serundeputy's [suggestion](https://github.com/backdrop/backdrop/pull/1774/files#r99502715) to account for the cases where there might be zero content types (as pointed out by @docwilmot). Core standard profile does ship with default content types, but other profiles might be bare.